### DISCUSSION TITLE
People search returns Guest, blocked, unlicensed users, etc.

### DISCUSSION THREAD

**Discussion by OzmeisterO** [microsoft-search/pnp-modern-search#3074] [Q&A] [answered]
The People search is incredibly powerful and useful, but by default it returns ALL users including Guests, unlicensed users, blocked users, etc.  It would be REALLY helpful if there was an easy way to filter the results to configure whether to show these people or not.  I suspect most people would only want to show licensed users so that would be a great starting point.
If there's a query string I can use please let me know what it is and where to put it as I've tried various things and come up blank!
Thanks.

**Comment by kasperbolarsen** [comment, marked as answer]
perhaps you will find this usefull: https://ms365thinking.blogspot.com/2023/05/sharepoint-people-search-how-to-clean.html + https://www.techmikael.com/2018/01/excluding-sharepoint-online-service.html

**Reply by OzmeisterO** [reply]
Thanks @kasperbolarsen, very useful.  Unfortunately I didn't get anywhere though.  I tried putting the following in the 'Refinement filters' box of the PnP - Search Results web part:

1. "SPS-HideFromAddressLists"<>1 : Returned 0 results
2. -preferredname:test* : Returned 0 results
3. WorkEmail:@ourdomain.co : Did remove external users from the results

So I had very slight success, but not a great improvement unfortunately.  Are you able to clarify exactly what code I should put where please to only show current, licensed users?  I suspect it should be a combination of 1 & 3 above, but I couldn't get 1 to work  :-(

Thanks again, Oz

**Reply by kasperbolarsen** [reply]
For 1 and 2 you should be looking at the number of accounts returned as they exclude accounts, not returning them.
If you do a * search against the LocalPeopleResults you might get 1200 hits
Change * to "SPS-HideFromAddressLists"<>1 might yield 1150 hits as 50 accounts did have the value 1 in that property

You will have to conbine the criteria , like this one :

{searchTerms} "SPS-HideFromAddressLists"<>1 -AccountName:admin* -AccountName:test* -AccountName:EXT* WorkEmail:@contoso -Department: External 

In general I recommend using the SharePoint Search Query tool and/or the SP Editor chrome extention when hamering out the details in the query

**Reply by OzmeisterO** [reply]
Thanks for another prompt reply Kasper.  If the Refinement filters box of the PnP - Search Results web part is empty then I see 215 results in the web part.  My issue is that as soon as I add   {searchTerms} "SPS-HideFromAddressLists"<>1   or   "SPS-HideFromAddressLists"<>1   in the Refinement filters box, the web part shows 0 results.  So I'm guessing I'm either entering the query in the wrong place, or it's in the wrong format.

**Reply by OzmeisterO** [reply]
Ah, have just tried the following in the 'Query template' box and it appears to be working!:

{searchTerms} "SPS-HideFromAddressLists"<>1 WorkEmail:@contoso -AccountName:admin* -AccountName:test*

Thanks again for all your help.

**Reply by OzmeisterO** [reply]
Just as a quick follow-up after further testing, for some reason even though I've included "SPS-HideFromAddressLists"<>1 it is still returning users whose accounts are hidden from the GAL.  Strange.

**Reply by kasperbolarsen** [reply]
Well, it all depends on the value in the User Profile properties.  Perhaps the property isn't synchronized ?
Can you check the value in the SharePoint Search Query tool or SP Editor?

**Comment by mw-new-test** [comment]
We have a similar requirement.
SharePoint people results are returning accounts that are unlicensed and even service accounts that are not really managed/owned by a user.
Is there a way to exclude these results and only return People?
We currently have a workaround to exclude accounts that don't really have a Department/Job Title which is not an ideal solution.
We cannot always consider that unlicensed users and service accounts may not have Departments/JobTitles.

Any help is appreciated.

**Reply by kasperbolarsen** [reply]
The results in you people search is based on the information in your User Profiles which again are synchronized from AD/AAD, so the responsibility is yours, not Microsofts ;-) 
That being said, have you reviewed the options in the accepted answer? I have updated people search for a lot of our customers using those criteria and so far they have done the job.

**Reply by mw-new-test** [reply]
Thanks @kasperbolarsen.
We are currently using the same query suggested as answer.
We are also checking for Department of the user assuming that service accounts/admin accounts may not have a Departament.
But I don't think null check is possible in KQL - Let me know if it is or if Regex is supported in KQL.

Reallly don't like applying so many filters in query template.
{searchTerms} -"SPS-HideFromAddressLists":1 -AccountName:admin* -AccountName:ZZ* -AccountName:aad* -AccountName:ctl* -AccountName:adm* -AccountName:test* -AccountName:EXT* AND (Department:a* OR Department:b* OR Department:c* OR Department:d* OR Department:e* OR Department:f* OR Department:g* OR Department:h* OR Department:i* OR Department:j* OR Department:k* OR Department:l* OR Department:m* OR Department:n* OR Department:o* OR Department:p* OR Department:q* OR Department:r* OR Department:s* OR Department:t* OR Department:u* OR Department:v* OR Department:w* OR Department:x* OR Department:y* OR Department:z*

We are also in touch with our AD team to have a property in Active directory that will allow us to differentiate between accounts of people and accounts (like service accounts) not owned by any user.
